streaming_data_types/arrays_wa00.py

```diff
@@ -0,0 +1,144 @@
+from streaming_data_types.fbschemas.arrays_wav_00.WaveFormArray import (
+    WaveFormArrayStart,
+    WaveFormArrayAddXData,
+    WaveFormArrayAddYData,
+    WaveFormArrayAddXDataType,
+    WaveFormArrayAddYDataType,
+    WaveFormArrayAddYTimestamp,
+    WaveFormArrayAddXTimestamp,
+    WaveFormArrayAddYUnit,
+    WaveFormArrayAddXUnit,
+    WaveFormArrayAddNumberOfElements,
+    WaveFormArrayEnd,
+)
+from streaming_data_types.utils import check_schema_identifier
+
+FILE_IDENTIFIER = b"wa00"
+
+
+def serialise_wa00(
+    values_x_array: np.ndarray,
+    values_y_array: np.ndarray,
+    y_timestamp: Optional[datetime] = None,
+    x_timestamp: Optional[datetime] = None,
+    y_unit: Optional[str] = None,
+    x_unit: Optional[str] = None,
+    number_of_elements: Optional[np.uint32] = None
+) -> bytes:
+    builder = flatbuffers.Builder(1024)
+
+    type_map = {
+        np.dtype("uint8"): DType.uint8,
+        np.dtype("int8"): DType.int8,
+        np.dtype("uint16"): DType.uint16,
+        np.dtype("int16"): DType.int16,
+        np.dtype("uint32"): DType.uint32,
+        np.dtype("int32"): DType.int32,
+        np.dtype("uint64"): DType.uint64,
+        np.dtype("int64"): DType.int64,
+        np.dtype("float32"): DType.float32,
+        np.dtype("float64"): DType.float64,
+    }
+
+
+    datatype_x_array = type_map[values_x_array.dtype]
+    datatype_y_array = type_map[values_y_array.dtype]
+
+    # Build data
+    x_data_offset = builder.CreateNumpyVector(values_x_array.view(np.uint8))
+    y_data_offset = builder.CreateNumpyVector(values_y_array.view(np.uint8))
+    if y_unit is not None:
+        y_unit_offset = builder.CreateString(y_unit)
+    if x_unit is not None:
+        x_unit_offset = builder.CreateString(x_unit)
+
+    # Build the buffer
+    WaveFormArrayStart(builder)
+    WaveFormArrayAddXData(builder, x_data_offset)
+    WaveFormArrayAddYData(builder, y_data_offset)
+    WaveFormArrayAddXDataType(builder, datatype_x_array)
+    WaveFormArrayAddYDataType(builder, datatype_y_array)
+    if y_timestamp is not None:
+        WaveFormArrayAddYTimestamp(builder, int(y_timestamp.timestamp() * 1e9))
+    if x_timestamp is not None:
+        WaveFormArrayAddXTimestamp(builder, int(x_timestamp.timestamp() * 1e9))
+    if y_unit is not None:
+        WaveFormArrayAddYUnit(builder,y_unit_offset)
+    if x_unit is not None:
+        WaveFormArrayAddXUnit(builder,x_unit_offset)
+    if number_of_elements is not None:
+        WaveFormArrayAddNumberOfElements(builder, number_of_elements)
+
+    WA_Message = WaveFormArrayEnd(builder)
+    builder.Finish(WA_Message, file_identifier=FILE_IDENTIFIER)
+    return bytes(builder.Output())
+
+@dataclass
+class wa00_t:
+    values_x_array: np.ndarray
+    values_y_array: np.ndarray
+    y_timestamp: Optional[np.uint64] = None
+    x_timestamp: Optional[np.uint64] = None
+    x_unit: Optional[str] = None
+    y_unit: Optional[str] = None
+    number_of_elements: Optional[np.uint32] = None
```

Comment:
I don't think there's any need to include this. It was intended to help a programmer reshape the arrays. Since you've already done that here, there's no point including it.